### PR TITLE
feat: add optional questions

### PR DESCRIPTION
**Pull request by valerydluski**

### CODE REVIEW

**Comment by valerydluski** [comment]
@helenakrasnova

File: tasks/interview-corejs.md [hunk 1]
```diff
@@ -65,29 +65,29 @@
...
 
-  - Immediately invoked functional expression `(IIFE)`
-  - Know IIFE pattern
+  - Immediately invoked functional expression `(IIFE)` `(optional)`
+  - Know IIFE pattern `(optional)`
   - Callback (Function as argument)
   - Know callback pattern
-  - Understand callback limitations (callback hell)
+  - Understand callback limitations (callback hell) `(optional)`
   - Binding, binding one function twice
```
Comment (helenakrasnova) [inline tasks/interview-corejs.md:73]: Maybe it should be divided into 2 questions: 1)binding 2) binding one function twice and make second question as optional

File: tasks/interview-corejs.md [hunk 2]
```diff
@@ -165,8 +165,8 @@
   - Know browser History APIconcept
   - Be able to navigate within browser history
   - Be able to use history state
```
Comment (helenakrasnova) [inline tasks/interview-corejs.md:167]: It should be optional, I suppose

File: tasks/interview-corejs.md [hunk 3]
```diff
@@ -116,11 +116,11 @@
...
   - Able to use [Object.create] and define `__proto__` explicitly
-  - Able to set / get object prototype
+  - Able to set / get object prototype 
   - `prototype` property
   - Know function `prototype` property
   - Understand dependency between function constructor `prototype` and instance `__proto__`
-  - Able to create 'class' methods using function `prototype` property
+  - Able to create 'class' methods using function `prototype` property 
 
 - #### ECMAScript Classes
```
Comment (helenakrasnova) [inline tasks/interview-corejs.md:125]: make Getter/setter question as optional?

File: tasks/interview-corejs.md [hunk 4]
```diff
@@ -165,8 +165,8 @@
   - Know browser History APIconcept
   - Be able to navigate within browser history
   - Be able to use history state
-  - Navigator
-  - Know how to parse user agent
+  - Navigator `(optional)`
+  - Know how to parse user agent `(optional)`
   - Know how to discover client platform, browser
   - Cookies
```
Comment (helenakrasnova) [inline tasks/interview-corejs.md:171]: cookies/localStorage/sessionStorage

File: tasks/interview-corejs.md [hunk 5]
```diff
@@ -210,8 +210,8 @@
   - using interfaces with optional properties, read-only properties, etc...
   - creating custom types.
```
Comment (helenakrasnova) [inline tasks/interview-corejs.md:211]: + types/interface differences (optional)

File: tasks/interview-corejs.md [hunk 6]
```diff
@@ -220,18 +220,18 @@
 
   - design patterns used on his project, and able to compare these patterns
```
Comment (helenakrasnova) [inline tasks/interview-corejs.md:221]: and JavaScript Design Patterns (optional) -?

**Pull request merged**